cmake/SetupThirdPartyTargetsDownstream.cmake

```diff
@@ -0,0 +1,28 @@
+# This file is intended to be included in the *-config.cmake files of
+# any project using a third-party library.  The macro 
+# `blt_install_tpl_setups(DESTINATION <dir>)`  installs this file
+# into the destination specified by the argument <dir>.
+
+# Consider an example project foo, built using BLT, that depends on OpenMP.  
+# This file is used by projects that link to foo, to create a target for OpenMP
+# with valid config flags. 
+include("${CMAKE_CURRENT_LIST_DIR}/BLTOptions.cmake")
+# BLTInstallableMacros provides helper macros for setting up and creating
+# third-party library targets.
+include("${CMAKE_CURRENT_LIST_DIR}/BLTInstallableMacros.cmake")
+
+if (EXISTS "${CMAKE_CURRENT_LIST_DIR}/SetupMPI.cmake")
+    include("${CMAKE_CURRENT_LIST_DIR}/SetupMPI.cmake")
+endif()
```

Comment:
Either inside these files, or here - we need a guard to prevent their logic from being executed again if they were already included by BLT proper. 
If not, the MPI search logic, etc will be shown twice for BLT folks.

Reply:
good point!